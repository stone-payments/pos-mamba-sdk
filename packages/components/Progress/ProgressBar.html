<div class="progress-bar {progressClass}" {style}>
  {#if typeof progress === undefined}
    <div class="progress"></div>
  {:else}
    <div class="progress" style="width: {progress}%;"></div>
  {/if}
</div>

<script>
  export default {
    data() {
      return {
        color: '#6ebf1a',
        height: '6px',
        progress: undefined,
      };
    },
    computed: {
      progressClass({ progress }) {
        return typeof progress === 'undefined' ? 'is-infinite' : '';
      },
      style({ color, height }) {
        return [`background-color:${color}`, `height:${height}`].join(';');
      },
    },
  };
</script>

<style type="text/postcss">
  @import '@mamba/styles/colors.pcss';

  .progress-bar {
    display: block;
    width: 100%;
    overflow: hidden;
    background-color: $green;
  }

  .progress {
    height: inherit;
    background-color: #000;
    opacity: 0.3;

    .progress-bar.is-infinite & {
      transform: translateX(-100%);
      width: 80%;
      animation: indeterminate 2.1s ease-in-out infinite;
    }
  }

  @keyframes indeterminate {
    to {
      transform: translateX(125%);
    }
  }
</style>
