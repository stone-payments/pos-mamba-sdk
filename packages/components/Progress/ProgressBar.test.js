import ProgressBar from './ProgressBar.html';

const { newTestRoot } = global;

const root = newTestRoot();

const newProgressBar = data =>
  root.createComponent(ProgressBar, { unique: true, data });

it('should create the default component', () => {
  newProgressBar();

  expect(root.query('.progress-bar')).toBeTruthy();
  expect(root.query('.progress-bar.is-infinite')).not.toBeNull();
});

it('should match all params', () => {
  newProgressBar({
    progress: '30',
    height: '2px',
    color: 'blue',
  });

  const barEl = root.query('.progress-bar');

  expect(root.target.children[0]).not.toBeUndefined();

  expect(barEl).not.toBeUndefined();
  expect(barEl.children[0].style.width).toBe('30%');
  expect(barEl.style.height).toBe('2px');
  expect(barEl.style.backgroundColor).toBe('blue');
});

// Must do color/theme specific tests
describe('colors/theme', () => {
  it('should render bar with color', () => {
    newProgressBar({ color: 'red' });

    expect(root.query('.progress-bar').style.backgroundColor).toBe('red');
  });
});

// Must do bar size specific test
describe('sizes', () => {
  it('should render bar with height', () => {
    newProgressBar({ height: '12px' });

    expect(root.query('.progress-bar').style.height).toBe('12px');
  });
});

// Must do bar behavior specific tests
describe('custom progress', () => {
  it('should set bar progress', () => {
    newProgressBar({ progress: '85' });

    expect(root.query('.progress-bar').children[0].style.width).toBe('85%');
  });
});

describe('infinite bar', () => {
  it('infinite progress bar animation should start at -100%', () => {
    newProgressBar();

    const barEl = root.query('.progress-bar.is-infinite');

    expect(getComputedStyle(barEl.children[0]).transform).toBe(
      'translateX(-100%)',
    );
  });

  it('infinite progress bar width should have less than 100%', () => {
    newProgressBar();

    const barEl = root.query('.progress-bar.is-infinite');

    expect(getComputedStyle(barEl.children[0]).width).not.toBe('100%');
  });
});
