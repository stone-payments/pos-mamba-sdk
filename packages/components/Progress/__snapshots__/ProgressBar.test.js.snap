// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should create the default component 1`] = `
<div
  class="progress-bar is-infinite svelte-qns1so"
  style="background-color: rgb(110, 191, 26); height: 6px;"
>
  <div
    class="progress svelte-qns1so"
  />
</div>
`;
